### PR TITLE
REG: Fix read_parquet from file-like objects

### LINKED ISSUE #34467: BUG: read_parquet no longer supports file-like objects
#### Code Sample, a copy-pastable example

```python
from io import BytesIO
import pandas as pd

buffer = BytesIO()

df = pd.DataFrame([1,2,3], columns=["a"])
df.to_parquet(buffer)

df2 = pd.read_parquet(buffer)
```

#### Problem description

The current behavior of `read_parquet(buffer)` is that it raises the following exception:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "./working_dir/tvenv/lib/python3.7/site-packages/pandas/io/parquet.py", line 315, in read_parquet
    return impl.read(path, columns=columns, **kwargs)
  File "./working_dir/tvenv/lib/python3.7/site-packages/pandas/io/parquet.py", line 131, in read
    path, filesystem=get_fs_for_path(path), **kwargs
  File "./working_dir/tvenv/lib/python3.7/site-packages/pyarrow/parquet.py", line 1162, in __init__
    self.paths = _parse_uri(path_or_paths)
  File "./working_dir/tvenv/lib/python3.7/site-packages/pyarrow/parquet.py", line 47, in _parse_uri
    path = _stringify_path(path)
  File "./working_dir/tvenv/lib/python3.7/site-packages/pyarrow/util.py", line 67, in _stringify_path
    raise TypeError("not a path-like object")
TypeError: not a path-like object
```

#### Expected Output

Instead, `read_parquet(buffer)` should return a new DataFrame with the same contents as the serialized DataFrame stored in `buffer`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-99-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.4
numpy            : 1.18.4
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 9.0.1
setuptools       : 39.0.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 0.999999999
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.17.1
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] xref #34467
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry - waiting on https://github.com/pandas-dev/pandas/pull/34481

Use arrow parquet.read_table opposed to ParquetDataset

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
> whatsnew entry - waiting on #34481

I merged that PR

**Comment by jorisvandenbossche** [comment]
The original version (before https://github.com/pandas-dev/pandas/pull/33632) used `get_filepath_or_buffer`, which eg also enables to read from urls. Do you know if that is tested?

File: pandas/tests/io/test_parquet.py
Comment (shubh2u) [inline pandas/tests/io/test_parquet.py:575]: Don't checkin print statements?

Comment (alimcmaster1) [reply pandas/tests/io/test_parquet.py:575]: Yep thanks - I’m about to tidy this up.

**Comment by alimcmaster1** [comment]
> The original version (before #33632) used `get_filepath_or_buffer`, which eg also enables to read from urls. Do you know if that is tested?

Agree & thanks - it appears not to be tested let me add.

File: pandas/tests/io/test_parquet.py
Comment (alimcmaster1) [inline pandas/tests/io/test_parquet.py:572]: We should probably test this for fastparquet too - seems not to work currently

File: pandas/io/parquet.py
Comment (alimcmaster1) [inline pandas/io/parquet.py:127]: We have some similar logic on the fastparquet side. Should consolidate in the future: https://github.com/pandas-dev/pandas/blob/master/pandas/io/parquet.py#L188

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good to me. 
The url still needs to be changed before merging?

Do you think this should be safe to use for 1.0.5? (as the other option is to revert the original PR for 1.0.5, and keep this for master only)

cc @simonjayhawkins

**Review by jreback — APPROVED** [review]
lgtm. i think ok for 1.0.5

File: pandas/tests/io/test_parquet.py
Comment (martindurant) [inline pandas/tests/io/test_parquet.py:576]: This might fail due to rate limits from github?

Comment (alimcmaster1) [reply pandas/tests/io/test_parquet.py:576]: Yep fair point - we already do this in test_network.py and I think the decorator helps handle any failures. We could use https://pypi.org/project/pytest-localserver/ ? also I couldnt find docs that suggest what the rate limits are for raw.githubusercontent endpoints?

**Comment by alimcmaster1** [comment]
> Looks good to me.
> The url still needs to be changed before merging?
> 
> Do you think this should be safe to use for 1.0.5? (as the other option is to revert the original PR for 1.0.5, and keep this for master only)
> 
> cc @simonjayhawkins

I was planning on updating the URL post merge. Other option is I can open a separate PR with just .parquet file so it exists on master. I think should be fine for 1.0.5 - any additional test cases you can think of would be helpful. I’ll take a look at current coverage.

**Comment by jorisvandenbossche** [comment]
I would propose to keep this for 1.1 (and we reverted to original patch in the 1.0.x branch for 1.0.5). @alimcmaster1 you can remove the whatsnew note then? (we still need to add a similar line to the v1.0.5.txt, but that should be done in a separate PR)

**Comment by alimcmaster1** [comment]
> I would propose to keep this for 1.1 (and we reverted to original patch in the 1.0.x branch for 1.0.5). @alimcmaster1 you can remove the whatsnew note then? (we still need to add a similar line to the v1.0.5.txt, but that should be done in a separate PR)

Yes makes sense - I’ll do this tomorrow.

**Comment by jreback** [comment]
this now doesn't close the issue as that's actually marked for 1.0.5? 

The plan is to patch that separately right?

**Comment by TomAugspurger** [comment]
IIUC, all that needs to be done is move the release note to 1.1.0.rst. I'll do that now.

**Comment by TomAugspurger** [comment]
Actually, I'm sufficintly confused about what the appropriate whatsnew to describe the changes from 1.0.5 to 1.1 is, so I'll leave that to you @alimcmaster1.

**Comment by jorisvandenbossche** [comment]
In this PR, the whatsnew only needs to be simply removed (this is fixing a regression compared to master, so doesn't need a whatsnew). Describing what's changed in 1.0.5, that's for a separate PR that gets backported.

Will do that now

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
I also updated the URL to point to the master branch, so this is going to fail anyway here, thus merging directly

**Comment by jorisvandenbossche** [comment]
Thanks @alimcmaster1 !

**Comment by alimcmaster1** [comment]
Thanks for fixing up @jorisvandenbossche - apologies I didn’t get to this. I’ll add the 1.0.5 whatsnew note you mentioned

**Pull request merged**
